src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourceWindowManager.java

```diff
@@ -209,14 +209,14 @@ public void onBlur(BlurEvent arg0)
          }
       }
       
+      // signal that this window has focus
+      MainWindowObject.lastFocusedWindow().set(getSourceWindowId());
```

Comment:
Do we need to check to see if the window has focus here? (consider e.g. if user opens a source window, but switches back to the main window before the source window has loaded its scripts)